### ISSUE TITLE
pdfview buffer isn't reverted when exporting from org

### ISSUE THREAD

**Issue by dinojr** [politza/pdf-tools#172]
I'm using PDFView to display a pdf file produced by the export process of orgmode. Even though auto-revert-mode is set in the buffer, I have to press g to revert the pdfview buffer when I update the pdf file from the orgmode buffer. If I compile the corresponding tex file, the pdf buffer is reverted as it should though.

**Comment by NLisa** [comment]
I have the same experience.

With `auto-revert-mode` enabled for the pdf file and `global-auo-revert-mode` turned-on, compiling tex file directly reverts the pdf buffer. Exporting from org file queries whether to re-read file from disk.

**Comment by NLisa** [comment]
The problem is with `org-export-dispatch` which calls `org-open-files` which in turn calls `find-file-other-window` which itself causes the problem by calling `find-file-noselect` with its `nowarn` argument set to _nil_

This issue should probably be closed as it is not a pdf-tools issue, but a problem with `org-export` not respecting `auto-revert-mode` / `global-auto-revert-move` settings.

**Comment by politza** [comment]
Note, that auto-reverting a PDF file does not work reliably.  Emacs may revert the buffer while it is still written (e.g. by LaTeX), resulting in an error.

What is the relevance behind calling `find-file-noselect` without `nowarn` ?

**Comment by NLisa** [comment]
I've posed the question on [emacs.stackexchange](http://emacs.stackexchange.com/questions/31520/org-export-does-not-respect-global-auto-revert-mode), but from what I can tell, without `nowarn`, I think it does mandatory checks that prompt the user, to test whether the file's buffer has changed or whether the file has changed on disk.

I believe that in the case where the file has changed on disk, there should be a check to see if `(global-)auto-revert-mode` in set, and if so automatically (without prompt) revert the file.

**Comment by politza** [comment]
From what I read, `nowarn` means that the buffer is reverted silently, i.e. without asking the user for confirmation.

**Comment by NLisa** [comment]
Sorry if I wasn't clear, thanks for succinctly phrasing what I'd intended to say. Yes, with `nowarn` set, buffer is silently reverted **and** changes to a visited buffer are silently discarded.

Very peculiar though is that the intermediary _.tex_  file that is generated is silently reverted even when you edit the _tex_ buffer, and its edits are silently discarded.

**Comment by politza** [comment]
I think, I got it now. The problem is that org indirectly calls `find-file-noselect` on an already visited PDF file and while the file has new contents, the buffer is not reverted to reflect these changes.  Is that right ?

I think that's how this function is supposed to work. (Maybe it would be a good idea to auto-revert it in the appropriate cases.)

**Comment by NLisa** [comment]
Might have not been such a good idea to hijack someone else's issue.

> Is that right ?

Yes. This is the case is @dinojr case, and I believe the solution for him is his pdf build command. I.e. he probably needs to use latexmk or re-run pdflatex a number of times to update references, and update the timestamp on the pdf to ensure that `auto-revert-mode` is initiated on the latest file:

    (setq org-latex-pdf-process 
       '("latexmk -pdflatex='%latex -shell-escape -interaction nonstopmode' -pdf -output-directory=%o -f %f"))

or

    (setq org-latex-pdf-process
       '("pdflatex -interaction nonstopmode -output-directory %o %f"
 	  "biber %b"
 	  "pdflatex -interaction nonstopmode -output-directory %o %f"
 	  "pdflatex -interaction nonstopmode -output-directory %o %f"))

In my particular case, I am prompted with a `yes-or-no-p` statement whether I want to revert the buffer as it has changed on disk, even though I have set `(global-)auto-revert-mode` for _that_ buffer. I think I gave a slightly clearer explanation in my [Emacs.StackExchange Question on this Matter](http://emacs.stackexchange.com/questions/31520/org-export-does-not-respect-global-auto-revert-mode), and quoted what I believe to be the offending code in _emacs/lisp/files.el.gz_, where it was suggested that I instead look to modify `org-export-dispatch`, so that the cascading calls to `org-open-files` -> `find-files-new-window` -> `find-files-noselect` pass along the `nowarn` argument.

Anyway, I think since @dinojr is not being prompted as I am in my case, he may just have to run pdflatex multiple times. So this issue can probably be closed once he can confirm that multiple calls to pdflatex in his `org-latex-pdf-process` solves this issue. I'll check my `org-file-apps`, I think the way I'm _opening_ pdfs by connecting to an emacsclient daemon may be giving issues with regards to _stealing_ an open file and that's why I'm being prompted.
